### PR TITLE
Specify custom signer scope for morph invocations

### PR DESCRIPTION
**Pull request by fyrchik**
Related #749 .

### CODE REVIEW

File: cmd/neofs-adm/internal/modules/morph/group.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	"github.com/nspcc-dev/neo-go/cli/input"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/manifest"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/spf13/viper"
+)
+
+const contractWalletName = "contract.json"
```
Comment (cthulhu-rider) [inline cmd/neofs-adm/internal/modules/morph/group.go:16]: To be more clear I'd name it
```suggestion
const contractWalletFilename = "contract.json"
```

File: cmd/neofs-adm/internal/modules/morph/group.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+func openManifestWallet(walletDir string) (*wallet.Wallet, error) {
+	p := path.Join(walletDir, contractWalletName)
+	w, err := wallet.NewWalletFromFile(p)
+	if err != nil {
+		return nil, fmt.Errorf("can't open wallet: %w", err)
+	}
+
+	var password string
+	if key := "credentials.contract"; viper.IsSet(key) {
```
Comment (cthulhu-rider) [inline cmd/neofs-adm/internal/modules/morph/group.go:66]: Seems similar to code block in `initializeManifestWallet`, maybe unify them?
Btw `key` can be `const`.

**Review by cthulhu-rider — APPROVED** [review]

File: cmd/neofs-adm/internal/modules/morph/group.go (hunk 2 above)
Comment (fyrchik) [reply cmd/neofs-adm/internal/modules/morph/group.go]: Fixed.

File: cmd/neofs-adm/internal/modules/morph/group.go (hunk 1 above)
Comment (fyrchik) [reply cmd/neofs-adm/internal/modules/morph/group.go]: Fixed

File: pkg/morph/client/nns.go [hunk 3]
```diff
@@ -143,3 +166,46 @@ func exists(c *client.Client, nnsHash util.Uint160, domain string) (bool, error)
 	// and, therefore, exists
 	return !available, nil
 }
+
+// SetGroupSignerScope makes the default signer scope include all NeoFS contracts.
+// Should be called for side-chain client only.
+func (c *Client) SetGroupSignerScope() error {
```
Comment (alexvanin) [inline pkg/morph/client/nns.go:172]:
```diff
 // SetGroupSignerScope makes the default signer scope include all NeoFS contracts.
 // Should be called for side-chain client only.
-func (c *Client) SetGroupSignerScope() error {
+func (c *Client) SetGroupSignerScope() (err error) {
+       if c.multiClient != nil {
+               return c.multiClient.iterateClients(func(c *Client) error {
+                       // wrap error to stop client iteration
+                       return wrapNeoFSError(c.SetGroupSignerScope())
+               })
+       }
+
        pub, err := c.contractGroupKey()
        if err != nil {
                return err
```

**Review by alexvanin — CHANGES_REQUESTED** [review]
Tried on neofs-contract v0.14.2 with `group.neofs` NNS record and on v0.14.0 without NNS record. With requested fix works fine.

File: pkg/morph/client/nns.go [hunk 4]
```diff
@@ -58,26 +62,38 @@ func (c *Client) NNSContractAddress(name string) (sh util.Uint160, err error) {
...
+	sh, err = nnsResolve(c.client, nnsHash, name)
 	if err != nil {
 		return sh, fmt.Errorf("NNS.resolve: %w", err)
 	}
 	return sh, nil
 }
 
-func nnsResolve(c *client.Client, nnsHash util.Uint160, domain string) (util.Uint160, error) {
+// NNSHash returns NNS contract hash.
+func (c *Client) NNSHash() (util.Uint160, error) {
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply pkg/morph/client/nns.go:78]: Fixed.

File: pkg/morph/client/nns.go (hunk 3 above)
Comment (fyrchik) [reply pkg/morph/client/nns.go:181]: Fixed.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**